### ISSUE TITLE
Error installing cryptography 3.4.6 

### ISSUE THREAD

**Issue by neumantm** [nix-community/poetry2nix#261]
When using `poetry2nix.mkPoetryApplication` with the following dependencies an error occurs:
```
[tool.poetry.dependencies]
python = "^3.7"
cryptography = "^3.1"
```

The full pyproject.toml, poetry.lock and default.nix are attached: [test.zip](https://github.com/nix-community/poetry2nix/files/6011198/test.zip)

The log output is:
```
$ nix-shell default.nix 
these derivations will be built:
  /nix/store/i9c2lpk4wim00di3n82vqdj60p7frjvc-python3.8-cryptography-3.4.6.drv
building '/nix/store/i9c2lpk4wim00di3n82vqdj60p7frjvc-python3.8-cryptography-3.4.6.drv'...
Sourcing python-recompile-bytecode-hook.sh
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pip-build-hook
Using pipBuildPhase
Using pipShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
unpacking sources
unpacking source archive /nix/store/z1dzq5f4m7g735mm0r9gpbi5bf5xph4c-cryptography-3.4.6.tar.gz
source root is cryptography-3.4.6
setting SOURCE_DATE_EPOCH to timestamp 1613513622 of file cryptography-3.4.6/setup.cfg
patching sources
configuring
no configure script, doing nothing
building
Executing pipBuildPhase
Removing pyproject.toml...
Creating a wheel...
WARNING: The directory '/homeless-shelter/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Processing /build/cryptography-3.4.6
    ERROR: Command errored out with exit status 1:
     command: /nix/store/0yhk4sk4x9s9hsrf3p1skbfy1pwd1rbf-python3-3.8.5/bin/python3.8 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/build/pip-req-build-b7jru7vh/setup.py'"'"'; __file__='"'"'/build/pip-req-build-b7jru7vh/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /build/pip-pip-egg-info-7v7od94t
         cwd: /build/pip-req-build-b7jru7vh/
    Complete output (14 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/build/pip-req-build-b7jru7vh/setup.py", line 14, in <module>
        from setuptools_rust import RustExtension
    ModuleNotFoundError: No module named 'setuptools_rust'

            =============================DEBUG ASSISTANCE==========================
            If you are seeing an error here please try the following to
            successfully install cryptography:

            Upgrade to the latest pip and try again. This will fix errors for most
            users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
            =============================DEBUG ASSISTANCE==========================

    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
builder for '/nix/store/i9c2lpk4wim00di3n82vqdj60p7frjvc-python3.8-cryptography-3.4.6.drv' failed with exit code 1
error: build of '/nix/store/i9c2lpk4wim00di3n82vqdj60p7frjvc-python3.8-cryptography-3.4.6.drv' failed


```

**Comment by adisbladis** [comment]
This should have been fixed in 61f3dc5e5412eee969d76bc119e993a7289e978b. Please reopen if this issue persists.

**Comment by samrose** [comment]
I am still seeing this issue

https://gist.github.com/samrose/0aa19431222c56275c2f77dcb0b803c3
